Fix string predicate parsing to handle cases with no parameters

eshan-savla opened the pull request:
This pull request includes a minor change to the `plansys2_pddl_parser` module. The change involves the `fromStringPredicate` function in the `Utils.cpp` file, adding checks to handle edge cases where the predicates have no parameters.

* [`plansys2_pddl_parser/src/plansys2_pddl_parser/Utils.cpp`](diffhunk://#diff-4673b16b2bf073681d59660e7d2d206f80c65307c5bad20774d5bb5f7a6305baR944-R951): Added a check to remove a closing parenthesis from the end of the first token and an early return if the token list size is one, indicating no parameters are provided.

eshan-savla committed:
Fix string predicate parsing to handle cases with no parameters

eshan-savla commented:
Would this also be considered a bug fix? If so, I can push the respective change from the humble branch I am working on

roveri-marco commented:
Can you also add a test case that was failing before the patch and that pass after the patch?

fmrico commented:
> Can you also add a test case that was failing before the patch and that pass after the patch?

Yes, please, @eshan-savla . A test will be very helpful to merge this PR

eshan-savla committed:
added test case for no arg predicates

eshan-savla committed:
fix

eshan-savla committed:
more fixes

eshan-savla committed:
Fix string predicate parsing to handle cases with no parameters

eshan-savla committed:
Merge branch 'fix_predicate_parsing' of https://github.com/eshan-savla/ros2_planning_system into fix_predicate_parsing

eshan-savla committed:
Revert "Fix string predicate parsing to handle cases with no parameters"

This reverts commit 1e4caeb539c143e69159cdb9b2e2f33aea956155.

eshan-savla committed:
Revert "Revert "Fix string predicate parsing to handle cases with no parameters""

This reverts commit 182c30f69f028b8c6c6ddbe8934209aaba21f0a9.

eshan-savla commented:
Hello all, I have added the requested test cases and rebased the fix on those commits. Now you should be able to also manually test whether these tests fail before my fix.

eshan-savla committed:
Merge branch 'PlanSys2:rolling' into fix_predicate_parsing

fmrico commented:
Thanks @eshan-savla 

Merging!!! :rocket:
